app/assets/javascripts/directives/git_references_typeahead_directive.js

```diff
@@ -0,0 +1,118 @@
+samson.directive('gitReferencesTypeahead', function() {
```

Comment:
@henders, had a few problems with using the existing Git references typeahead with AngularJS. AngularJS directives apparently run after the jQuery document is loaded, meaning, when the jQuery bindings run, the AngularJS templates have not been loaded yet and the target elements are still not in the DOM. For this reason, the typeahead was not being initialized.

Ended up creating a generic component (directive) that searches for the Git reference and delegates the action to the parent Controller (i.e., the parent Controller defines what to do when the search button is clicked).